src/Compilers/CSharp/Portable/Emitter/Model/PropertySymbolAdapter.cs

```diff
@@ -168,7 +168,7 @@ bool Cci.ISignature.ReturnValueIsByRef
             get
             {
                 CheckDefinitionInvariant();
-                return this.Type is ByRefReturnErrorTypeSymbol;
+                return this.RefKind == RefKind.Ref;
```

Comment:
How do we handle relative order of between custom modifiers and ByRef modifier in metadata?